.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,15 @@
+COMMENT: Complete this TEMPLATE and delete all "COMMENT:" lines.
+
+COMMENT: Non-trivial PRs must address an existing Issue.
+COMMENT: Choose one of:
+Changes to partially address #xxxx
+Changes to Close #xxxx
+Small change with no associated Issue.
+
+COMMENT: Check all requirements now or later. Mark the PR as a Draft if later.
+- [ ] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
+- [ ] Minimal number of commits with logically grouped changes (use interactive
+         rebase to tidy your branch history if necessary).
+- [ ] No off-topic changes changes (use other PRs for these).
+- [ ] Tests with adequate coverage included (unit and/or functional).
+- [ ] Entry added to the change log `CHANGES.md` for next release.
```

Comment:
The only problem with this is that PRs will likely conflict with one another...

Reply:
That does happen in pandas. Then the developers normally ask contributors to rebase the pull request. e.g. https://github.com/pandas-dev/pandas/pull/25103#pullrequestreview-224761435

A bit annoying for the contributor, but takes 2 minutes. And saves a lot of time for the developers. Guess it depends on whether the pros are worth for us.

Reply:
(I am happy to re-base. Actually pretty obsessed with re-basing..., but others may not be so happy.)

Reply:
I'm a massive rebaser.  Note recommendation to use it tidy your branch history - makes review easier too.

Reply:
(I'll take this as resolved in favor of adding change log entries).